python/cudf/cudf/core/dataframe.py

```diff
@@ -430,18 +430,67 @@ def mask(self, other):
             df[col]._column = df[col]._column.set_mask(boolbits)
         return df
 
-    def __setitem__(self, name, col):
-        """Add/set column by *name or DataFrame*
+    def __setitem__(self, arg, value):
+        """Add/set column by *arg or DataFrame*
         """
-        if isinstance(name, DataFrame):
+
+        if isinstance(arg, DataFrame):
+            # not handling set_item where arg = df & value = df
```

Comment:
Should we throw an exception in that case?

Reply:
Yup we should raise one. I was not sure about if we wanted to tackle it in this pr or not.

Will throw a exception here for a more graceful failure.

Reply:
If it's easy enough to handle I would handle it, but again we could do in a follow up PR.

Reply:
I am gonna raise an exception because `pandas` behavior is  slightly weird here too.  I will raise an exception for now. 

Like pandas behaves differently if i give it a bool series/vs dataframe with different names so will like to tackle this in the pr where we fix the  inconsistency 2. 

```
df_0 = pd.DataFrame({'a':[1,2,3],'b':[0,1,2]})
df_1 = pd.DataFrame({'a':[True,True,False],'b':[True,True,False]})
df_2 = pd.DataFrame({'x':[-10,-12,13],'y':[-11,-12,12]})

df_0[df_1] = df_2
df_0

	a	b
0	NaN	NaN
1	NaN	NaN
2	3.0	2.0


df_0[[True,True,False]] = df_2
df_0

	a	b
0	-10.0	-11.0
1	-12.0	-12.0
2	3.0	2.0

```

Reply:
Added exception with commit https://github.com/rapidsai/cudf/pull/3442/commits/bfe22ec21737808d8180d9ee4da492503998677c